### DISCUSSION TITLE
Programmatic scrolling does not work anymore?

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3589] [1. Help] [answered]
As mentioned already on [stackoverflow](https://stackoverflow.com/questions/76298324/how-to-debug-a-scroll-block-in-javascript), I'm looking for help debugging a stopped scrollbehaviour.

When the live preview is refreshed, it actually works instantly, when the last scrolling position before refresh (which is saved by the browser) was below the threshold. So, something that is happening on page init is blocking the scroll permanently, when not triggered manually with a button click or programmatically with a timeout of about 500ms.

The code for scrolling did work a few weeks before, but also with some older AlpineJS versions it doesn't seem to work anymore now.

The code in question is as follows: `x-intersect:leave.threshold.95="$refs.xyz.scrollIntoView()"`. Do you have any idea how to approach/solve this?

**Comment by ekwoka** [comment]
I have seen this issue before, but it is not an Alpine issue. It's a Browser/JS issue. When I had it, I just had code that goes just before it that scrolls the page 1 px down and then calls scroll into view.

For the alpine implementation, It would be helpful to see a better example of what you're actually trying to do, since this example of the x-intersect and a ref makes me think this could be architected better to not be concerned with this issue at all. Possible scroll snapping.

**Reply by simonferndriger** [reply]
Sure, do you need the whole code of the index.html or what would you need? Just checked, the behaviour works still fine with the latest version of edge...

**Reply by simonferndriger** [reply]
This is the desired scroll behaviour: https://www.loom.com/share/291290bca34945de9bd74bdf55512df8

I tried (CSS) scroll snapping, however, doesn't seem to work when I set `scroll-snap-type: y mandatory` on the `body` and `scroll-snap-align: start` on `main` which is where the thumbnails are (below the intro).

**Reply by ekwoka** [reply, marked as answer]
I believe the scroll snap needs to be on the html element, since technically here's the body is not scrolling, the document is.

**Reply by simonferndriger** [reply]
Awesome, it works! However, it is rather fast...maybe it can be slowed down, have to read a bit about this CSS feature

**Reply by ekwoka** [reply]
Native smooth scrolling is a fixed speed, sadly, but it would be the same as the scroll into view.

**Reply by simonferndriger** [reply]
I actually removed the CSS version, because scrolling was not smooth anymore, and sometimes it jumped to the top without any reason from the middle of the main area during scrolling. And on initial load, it scrolled down automatically. Seems that this CSS feature is a bit buggy...

**Reply by ekwoka** [reply]
Interesting. I've had no issues with it personally where I use it in production.

hopefully the doing a scroll of 1 pixel and then the scroll into view can help.

**Comment by simonferndriger** [comment]
It really is... I tried it out using `window.scrollBy(0, 1)` before calling `$refs.main.scrollIntoView()`. But, unfortunately, it didn't work. At least
it still works in Edge, so there is hope it will work in Chrome again in the future.

Am Di., 23. Mai 2023 um 06:27 Uhr schrieb Eric Kwoka <
***@***.***>:
